spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -608,6 +608,53 @@ public View createView(
         "Creating a view is not supported by catalog: " + catalogName);
   }
 
+  @Override
+  public View replaceView(
+      Identifier ident,
+      String sql,
+      String currentCatalog,
+      String[] currentNamespace,
+      StructType schema,
+      String[] queryColumnNames,
+      String[] columnAliases,
+      String[] columnComments,
+      Map<String, String> properties)
+      throws NoSuchViewException, NoSuchNamespaceException {
+    if (null != asViewCatalog) {
+      Schema icebergSchema = SparkSchemaUtil.convert(schema);
+
+      StringJoiner joiner = new StringJoiner(",");
+      Arrays.stream(queryColumnNames).forEach(joiner::add);
+
+      try {
+        Map<String, String> props =
+            ImmutableMap.<String, String>builder()
+                .putAll(Spark3Util.rebuildCreateProperties(properties))
+                .put("queryColumnNames", joiner.toString())
+                .build();
+
+        org.apache.iceberg.view.View view =
+            asViewCatalog
+                .buildView(buildIdentifier(ident))
+                .withDefaultCatalog(currentCatalog)
+                .withDefaultNamespace(Namespace.of(currentNamespace))
+                .withQuery("spark", sql)
+                .withSchema(icebergSchema)
+                .withLocation(properties.get("location"))
+                .withProperties(props)
+                .replace();
+        return new SparkView(catalogName, view);
+      } catch (org.apache.iceberg.exceptions.NoSuchNamespaceException e) {
+        throw new NoSuchNamespaceException(currentNamespace);
+      } catch (org.apache.iceberg.exceptions.NoSuchViewException e) {
+        throw new NoSuchViewException(ident);
```

Comment:
If this is thrown, then I think the Spark exec node should catch it and create the view. The `CREATE OR REPLACE` operation is supposed to be idempotent so it should not fail if the view is concurrently dropped.

This probably isn't a big deal, but it would be nice to handle it.

Reply:
this only exists due to the limited default implementation (which does a drop+create) in Spark's `replace()` as can be seen [here](https://github.com/apache/spark/blob/9657f970f2dd4f98b9c6e25870fffd87fb444fcc/sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java#L143). That being said, I don't think we need to handle `NoSuchViewException` in our implementation, since ours is idempotent. I went ahead and removed catching this here.

Reply:
Why don't we need to handle the exception here? Translating to the right Spark exception seems like something we should definitely do.

I realize that we already check whether the view exists, but if the view is dropped concurrently, it could still fail. It's an edge case, but the right thing to do is to catch and issue the create, right? CREATE OR REPLACE should not ever fail because the view doesn't exist.

Reply:
This isn't a blocker, so I'll merge this PR. We can follow up on this later.

Reply:
I'll follow-up on this